ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PointSchedule/Peers.hs

```diff
@@ -111,3 +116,36 @@ mkPeers h as =
     advs [a] = [("adversary", a)]
     advs _   = zip enumAdvs as
     enumAdvs = (\ n -> PeerId ("adversary " ++ show n)) <$> [1 :: Int ..]
+
+-- | Make a 'Peers' structure from the honest value and the other peers. Fail if
+-- one of the other peers is the 'HonestPeer'.
+mkPeers' :: a -> [Peer a] -> Peers a
+mkPeers' value prs =
+    Peers (Peer HonestPeer value) (Map.fromList $ dupAdvPeerId <$> prs)
+  where
+    -- | Duplicate an adversarial peer id; fail if honest.
+    dupAdvPeerId :: Peer a -> (PeerId, Peer a)
+    dupAdvPeerId (Peer HonestPeer _) = error "cannot be the honest peer"
+    dupAdvPeerId peer@(Peer pid _)   = (pid, peer)
+
+-- | Make a 'Peers' structure from a non-empty list of peers. Fail if the honest
+-- peer is not exactly once in the list.
```

Comment:
No, it does fail if it appears multiple times, because it will then pass a list to `mkPeers'` which will contain an honest peer still, and `mkPeers'` will fail on that.

Reply:
Looking back, it is not obvious; maybe we could add a comment for that.

Reply:
Oh, I see. Yeah, maybe a comment helps.